src/components/FinalPage/FinalPage.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { Box, Heading, Paragraph, Grommet } from 'grommet';
+
+import { colorRed } from '../../styles/variables';
+
+const theme = {
+  box: {
+    extend: {
+      background: 'url("/image/BinarycodeWithHearts.png") no-repeat',
+      'background-size': '80%'
+    }
+  }
+};
+
+const renderFinalMessage = location => {
+  if (location && location.state.success) {
+    return (
+      <Heading level="1" textAlign="center">
+        Thank you!
+      </Heading>
+    );
+  } else {
+    return (
+      <Paragraph color={colorRed} textAlign="center">
+        Error occured :( {location.state.error}
```

Comment:
What will happen here, when `location` will be undefined ;)? Just make the `location` property required for the `FinalPage` component.